Unit-01/03-templating/app.py

```diff
@@ -0,0 +1,36 @@
+from flask import Flask, render_template, request
+
+app = Flask(__name__)
+
+@app.route("/calculate")
+def calc():
+	return render_template("calc.html")
+
+@app.route("/math")
+def math():	
+	result = None
+	num1 = int(request.args.get('num1'))
+	num2 = int(request.args.get('num2'))
+
+	if request.args.get('calculation') == "add": 
+		result = "the sum is {}".format(num1 + num2)
+
+	elif request.args.get('calculation') == "subtract":
```

Comment:
consider more abstraction here; lots of duplicated code. I think I listed how to do this with a dict on the other PR, but here it is again:


```python
def add(a, b):
    return a + b
def subtract(a, b):
    return a - b
def multiply(a, b):
    return a * b
def divide(a / b):
    return a / b

operations = {
    "add": add,
    "subtract": subtract,
    "multiply": multiply,
    "divide": divide
}

# @route handler
def math(operation, num1, num2):
    if operation in operations:
        return f"{operations[operation](num1, num2)}"
```